bridges/scalajs-1/src/main/scala/bloop/scalajs/JsBridge.scala

```diff
@@ -147,25 +147,32 @@ object JsBridge {
       env: Map[String, String]
   ): (List[sbt.testing.Framework], () => Unit) = {
     implicit val debugFilter: DebugFilter = DebugFilter.Test
-    val nodeModules = baseDirectory.resolve("node_modules").toString
-    logger.debug("Node.js module path: " + nodeModules)
-    val fullEnv = Map("NODE_PATH" -> nodeModules) ++ env
-    val config =
-      NodeJSConfig().withExecutable(nodePath).withCwd(Some(baseDirectory)).withEnv(fullEnv)
-    val nodeEnv =
-      if (!jsConfig.jsdom.contains(true)) new NodeJSEnv(logger, config)
-      else new JsDomNodeJsEnv(logger, config)
-
-    // The order of the scripts mandates the load order in the JavaScript runtime
-    val input = jsConfig.kind match {
-      case ModuleKindJS.NoModule => Input.Script(jsPath)
-      case ModuleKindJS.CommonJSModule => Input.CommonJSModule(jsPath)
-      case ModuleKindJS.ESModule => Input.ESModule(jsPath)
-    }
+    val nodeModules = baseDirectory.resolve("node_modules")
+    if (nodeModules.toFile().exists()) {
+      logger.debug("Node.js module path: " + nodeModules.toString())
+      val fullEnv = Map("NODE_PATH" -> nodeModules.toString()) ++ env
+      val config =
+        NodeJSConfig().withExecutable(nodePath).withCwd(Some(baseDirectory)).withEnv(fullEnv)
+      val nodeEnv =
+        if (!jsConfig.jsdom.contains(true)) new NodeJSEnv(logger, config)
+        else new JsDomNodeJsEnv(logger, config)
+
+      // The order of the scripts mandates the load order in the JavaScript runtime
+      val input = jsConfig.kind match {
+        case ModuleKindJS.NoModule => Input.Script(jsPath)
+        case ModuleKindJS.CommonJSModule => Input.CommonJSModule(jsPath)
+        case ModuleKindJS.ESModule => Input.ESModule(jsPath)
+      }
 
-    val testConfig = TestAdapter.Config().withLogger(new Logger(logger))
-    val adapter = new TestAdapter(nodeEnv, Seq(input), testConfig)
-    val result = adapter.loadFrameworks(frameworkNames).flatMap(_.toList)
-    (result, () => adapter.close())
+      val testConfig = TestAdapter.Config().withLogger(new Logger(logger))
+      val adapter = new TestAdapter(nodeEnv, Seq(input), testConfig)
+      val result = adapter.loadFrameworks(frameworkNames).flatMap(_.toList)
+      (result, () => adapter.close())
+    } else {
+      logger.error(
+        s"Cannot discover test frameworks, missing node_modules in test project, expected them at $nodeModules"
```

Comment:
Just trying to understand the full picture here. In the initial metals report about the tests lenses not showing up, that's because there was no `node_modules`? If that is the case, will this error float up and actually be shown to the user on the client side?

Reply:
To be honest, I don't know enough about how the node stuff works to really answer that. Before the process never finished so we got nothing back. In this situation would it actually fix the original issue and still find the necessary frameworks?

Reply:
No, since there is no node_modules, but since Metals doesn't run Scala JS tests currently, we wouldn't show them anyway. So this will just show the error. I don't fully understand when the `node_modules` should show up in that build, but we shouldn't break Metals on that for sure